### ISSUE TITLE
Incorrect HDR playback

### ISSUE THREAD

**Issue by CrushedAsian255** [iina/iina#5151]
**System and IINA version:** 

- macOS  `macOS Sonoma 14.5 (23F79)`
- IINA  `1.3.5 Build 141` `mpv 0.35.0-419-gf79458476b-dirty` `FFmpeg 6.0`

**Expected / Actual behavior:**
When playing [this HDR video](https://4kmedia.org/sony-whale-in-tonga-hdr-uhd-4k-demo/), at 22s, the image is extremely clipped. The video plays fine in QuickTime Player.

The device is a MacBook Pro 16 inch M3 Max

<details>
<summary><b>mpv log:</b></summary>

```
[   0.000][d][global] user path: '~/Library/Logs/com.colliderli.iina/2024-08-20-16-43-43_Hi2K0k/mpv.log' -> '~/Library/Logs/com.colliderli.iina/2024-08-20-16-43-43_Hi2K0k/mpv.log'
[   0.006][v][cplayer] mpv 0.35.0-419-gf79458476b-dirty Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
[   0.006][v][cplayer]  built on Tue May 30 08:54:21 2023
[   0.006][v][cplayer] FFmpeg version: 6.0
[   0.006][v][cplayer] FFmpeg library versions:
[   0.006][v][cplayer]    libavutil       58.2.100
[   0.006][v][cplayer]    libavcodec      60.3.100
[   0.006][v][cplayer]    libavformat     60.3.100
[   0.006][v][cplayer]    libswscale      7.1.100
[   0.006][v][cplayer]    libavfilter     9.3.100
[   0.006][v][cplayer]    libswresample   4.10.100
[   0.006][v][cplayer] 
[   0.006][v][cplayer] Configuration: -Dhtml-build=disabled -Djavascript=enabled -Dlibmpv=true -Dlua=luajit -Dlibarchive=enabled -Duchardet=enabled -Dlibbluray=enabled -Dcplayer=false -Dswift-build=disabled -Dmacos-cocoa-cb=disabled -Dmacos-media-player=disabled -Dmacos-touchbar=disabled -Dmanpage-build=disabled -Dprefix=/opt/homebrew/Cellar/mpv-iina/HEAD-f794584 -Ddatadir=/opt/homebrew/Cellar/mpv-iina/HEAD-f794584/share/mpv-iina -Dlibdir=/opt/homebrew/Cellar/mpv-iina/HEAD-f794584/lib -Dsysconfdir=/opt/homebrew/etc/mpv-iina -Dbuildtype=release -Dwrap_mode=nofallback
[   0.006][v][cplayer] List of enabled features: av-channel-layout bsd-fstatfs build-date cocoa coreaudio cplugins darwin ffmpeg gl gl-cocoa glob glob-posix gpl iconv javascript jpeg jpegxl lcms2 libarchive libass libavdevice libbluray libdl libm luajit osx-thread-name posix posix_shm rubberband rubberband-3 stdatomic threads uchardet vector videotoolbox-gl zimg zlib
[   0.006][v][cplayer] mpv 0.35.0-419-gf79458476b-dirty Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
[   0.006][v][cplayer]  built on Tue May 30 08:54:21 2023
[   0.006][v][cplayer] FFmpeg version: 6.0
[   0.006][v][cplayer] FFmpeg library versions:
[   0.006][v][cplayer]    libavutil       58.2.100
[   0.006][v][cplayer]    libavcodec      60.3.100
[   0.006][v][cplayer]    libavformat     60.3.100
[   0.006][v][cplayer]    libswscale      7.1.100
[   0.006][v][cplayer]    libavfilter     9.3.100
[   0.006][v][cplayer]    libswresample   4.10.100
[   0.006][v][cplayer] 
[   0.006][v][cplayer] Configuration: -Dhtml-build=disabled -Djavascript=enabled -Dlibmpv=true -Dlua=luajit -Dlibarchive=enabled -Duchardet=enabled -Dlibbluray=enabled -Dcplayer=false -Dswift-build=disabled -Dmacos-cocoa-cb=disabled -Dmacos-media-player=disabled -Dmacos-touchbar=disabled -Dmanpage-build=disabled -Dprefix=/opt/homebrew/Cellar/mpv-iina/HEAD-f794584 -Ddatadir=/opt/homebrew/Cellar/mpv-iina/HEAD-f794584/share/mpv-iina -Dlibdir=/opt/homebrew/Cellar/mpv-iina/HEAD-f794584/lib -Dsysconfdir=/opt/homebrew/etc/mpv-iina -Dbuildtype=release -Dwrap_mode=nofallback
[   0.006][v][cplayer] List of enabled features: av-channel-layout bsd-fstatfs build-date cocoa coreaudio cplugins darwin ffmpeg gl gl-cocoa glob glob-posix gpl iconv javascript jpeg jpegxl lcms2 libarchive libass libavdevice libbluray libdl libm luajit osx-thread-name posix posix_shm rubberband rubberband-3 stdatomic threads uchardet vector videotoolbox-gl zimg zlib
[   0.006][d][global] home path: '' -> '-'
[   0.006][d][global] user path: '~~home/' -> ''
[   0.006][d][global] user path: '~~old_home/' -> ''
[   0.007][d][global] user path: '/Applications/IINA.app/Contents/Resources/config/iina-default-input.conf' -> '/Applications/IINA.app/Contents/Resources/config/iina-default-input.conf'
[   0.007][v][bdmv/bluray] Opening /Applications/IINA.app/Contents/Resources/config/iina-default-input.conf
[   0.007][v][file] Opening /Applications/IINA.app/Contents/Resources/config/iina-default-input.conf
[   0.007][d][file] resize stream to 1048576 bytes, drop 0 bytes
[   0.007][d][file] Stream opened successfully.
[   0.007][v][input] Parsing input config file /Applications/IINA.app/Contents/Resources/config/iina-default-input.conf
[   0.007][v][input] Input config file /Applications/IINA.app/Contents/Resources/config/iina-default-input.conf parsed: 72 binds
[   0.007][d][global] user path: '~/Library/Logs/com.colliderli.iina/2024-08-20-16-43-43_Hi2K0k/mpv.log' -> '~/Library/Logs/com.colliderli.iina/2024-08-20-16-43-43_Hi2K0k/mpv.log'
[   0.007][d][ytdl_hook] Loading lua script @ytdl_hook.lua...
[   0.007][d][stats] Loading lua script @stats.lua...
[   0.007][d][console] Loading lua script @console.lua...
[   0.007][d][auto_profiles] Loading lua script @auto_profiles.lua...
[   0.007][d][osx] Exiting...
[   0.007][d][stats] loading mp.defaults
[   0.007][v][cplayer] Waiting for scripts...
[   0.007][d][console] loading mp.defaults
[   0.007][d][ytdl_hook] loading mp.defaults
[   0.007][d][auto_profiles] loading mp.defaults
[   0.007][v][cplayer] Set property: vo="libmpv" -> 1
[   0.007][v][cplayer] Set property: keepaspect="no" -> 1
[   0.008][v][cplayer] Set property: gpu-hwdec-interop="auto" -> 1
[   0.008][d][stats] loading @stats.lua
[   0.008][d][auto_profiles] loading @auto_profiles.lua
[   0.008][d][console] loading @console.lua
[   0.008][d][ytdl_hook] loading @ytdl_hook.lua
[   0.008][d][ytdl_hook] reading options for ytdl_hook 
[   0.008][d][ytdl_hook] script-opts/ytdl_hook.conf not found. 
[   0.008][d][ytdl_hook] lua-settings/ytdl_hook.conf not found. 
[   0.008][d][stats] reading options for stats 
[   0.008][d][stats] script-opts/stats.conf not found. 
[   0.008][d][stats] lua-settings/stats.conf not found. 
[   0.067][v][ao/coreaudio] selected audio output device: MacBook Pro Speakers (109)
[   0.070][d][console] reading options for console 
[   0.070][d][console] script-opts/console.conf not found. 
[   0.070][d][console] lua-settings/console.conf not found. 
[   0.071][d][cplayer] Run command: define-section, flags=64, args=[name="input_console", contents="", flags="default"]
[   0.071][d][cplayer] Run command: enable-section, flags=64, args=[name="input_console", flags="allow-hide-cursor+allow-vo-dragging"]
[   0.071][d][cplayer] Run command: define-section, flags=64, args=[name="input_forced_console", contents="", flags="force"]
[   0.071][d][cplayer] Run command: enable-section, flags=64, args=[name="input_forced_console", flags="allow-hide-cursor+allow-vo-dragging"]
[   0.071][d][osd/libass] ASS library version: 0x1701000 (runtime 0x1701000)
[   0.071][v][osd/libass] libass API version: 0x1701000
[   0.071][v][osd/libass] libass source: tarball: 0.17.1
[   0.072][v][osd/libass] Shaper: FriBidi 1.0.12 (SIMPLE) HarfBuzz-ng 7.1.0 (COMPLEX)
[   0.072][v][osd/libass] Setting up fonts...
[   0.073][v][osd/libass] Using font provider coretext
[   0.073][v][osd/libass] Done.
[   0.073][d][cplayer] Run command: define-section, flags=64, args=[name="input_stats", contents="", flags="default"]
[   0.073][d][cplayer] Run command: enable-section, flags=64, args=[name="input_stats", flags="allow-hide-cursor+allow-vo-dragging"]
[   0.073][d][cplayer] Run command: define-section, flags=64, args=[name="input_forced_stats", contents="", flags="force"]
[   0.073][d][cplayer] Run command: enable-section, flags=64, args=[name="input_forced_stats", flags="allow-hide-cursor+allow-vo-dragging"]
[   0.073][d][auto_profiles] Exiting...
[   0.074][v][cplayer] Done loading scripts.

```
</details>

<br>

**Steps to reproduce:**
Download [this video](https://4kmedia.org/sony-whale-in-tonga-hdr-uhd-4k-demo/), and play it from the beginning up to 22 seconds with HDR enabled. The image will be extemely overblown.

- [x] MPV does not have this problem.

Video plays fine with mpv but mpv doesn't really support HDR playback, so the video is just SDR on mpv.

**How often does this happen?**
Every time this video is played

**Comment by low-batt** [comment]
What settings are you using for tone mapping in IINA's settings on the `Video/Audio` tab?

**Comment by CrushedAsian255** [comment]
> What settings are you using for tone mapping in IINA's settings on the `Video/Audio` tab?

Tone mapping is off, I'm playing the video back on the MacBook's internal HDR display.

Edit: tested different options, none help

**Comment by CrushedAsian255** [comment]
Huh, i think the MacBook's display just doesn't have enough headroom and was clipping the highlights, lowering the brightness has worked around the issue.

**Comment by low-batt** [comment]
So glad you reported that because this issue was really confusing me. I had downloaded that video and played it on my MacBook Pro with the M1 chip that as the XDR display like yours and did not see any clipping. I did not report that because I was still trying to think of what might be causing it to clip for you. I did not consider display headroom as an issue due to the greater capabilities of the XDR display compared to the displays in other Macs I own. 

I just checked my display settings and I have `Automatically adjust brightness` enabled and I'm usually testing in a room that is not very bright, so I would not have encountered an issue with headroom.
![display-settings](https://github.com/user-attachments/assets/9908c3cc-5ba8-48ce-8283-da8c9d263387)


Is is ok if I close this issue as solved?
